File: fstp-core-bo/resource/com/purefun/fstp/core/bo/otw/RDSStockBO_OTW.java
package com.purefun.fstp.core.bo.otw;

import com.purefun.fstp.core.bo.RDSStockBO;
import com.purefun.fstp.core.bo.pro.RDSStockBO_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fstp.core.bo.commom.ICommom_OTW;
import com.google.protobuf.Any;

public class RDSStockBO_OTW implements ICommom_OTW {
    RDSStockBO_PRO.RDSStockBO.Builder builder = null;
    RDSStockBO bo = null;

    public RDSStockBO_OTW() {
        builder = RDSStockBO_PRO.RDSStockBO.newBuilder();
        bo= new RDSStockBO();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public RDSStockBO_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = RDSStockBO_PRO.RDSStockBO.newBuilder();
        bo= new RDSStockBO();
        RDSStockBO_PRO.RDSStockBO receive = RDSStockBO_PRO.RDSStockBO.parseFrom(message);
        setProduct_id(receive.getProductId());
        setSecu_name_cn(receive.getSecuNameCn());
        setSecu_name_en(receive.getSecuNameEn());
        setExch_type(receive.getExchType());
        setSecu_type(receive.getSecuType());
        setSecu_sub_type(receive.getSecuSubType());
        setCurrency(receive.getCurrency());
        setList_date(receive.getListDate());
        setBuy_unit(receive.getBuyUnit());
        setSell_unit(receive.getSellUnit());
        setTrade_low_limit(receive.getTradeLowLimit());
        setTrade_high_limit(receive.getTradeHighLimit());
        setPre_close_price(receive.getPreClosePrice());
        setTick_price(receive.getTickPrice());
        setPrice_high_limit(receive.getPriceHighLimit());
        setPrice_low_limit(receive.getPriceLowLimit());
        setEx_right_ratio(receive.getExRightRatio());
        setDividend_price(receive.getDividendPrice());
        setFinancing_flag(receive.getFinancingFlag());
        setMargin_flag(receive.getMarginFlag());
        setSecu_status(receive.getSecuStatus());
        setMemo(receive.getMemo());
        setUpdate_time(receive.getUpdateTime());
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
    }

    public RDSStockBO_OTW(RDSStockBO bofrom){
        builder = RDSStockBO_PRO.RDSStockBO.newBuilder();
        bo= new RDSStockBO();
        setProduct_id(bofrom.product_id);
        setSecu_name_cn(bofrom.secu_name_cn);
        setSecu_name_en(bofrom.secu_name_en);
        setExch_type(bofrom.exch_type);
        setSecu_type(bofrom.secu_type);
        setSecu_sub_type(bofrom.secu_sub_type);
        setCurrency(bofrom.currency);
        setList_date(bofrom.list_date);
        setBuy_unit(bofrom.buy_unit);
        setSell_unit(bofrom.sell_unit);
        setTrade_low_limit(bofrom.trade_low_limit);
        setTrade_high_limit(bofrom.trade_high_limit);
        setPre_close_price(bofrom.pre_close_price);
        setTick_price(bofrom.tick_price);
        setPrice_high_limit(bofrom.price_high_limit);
        setPrice_low_limit(bofrom.price_low_limit);
        setEx_right_ratio(bofrom.ex_right_ratio);
        setDividend_price(bofrom.dividend_price);
        setFinancing_flag(bofrom.financing_flag);
        setMargin_flag(bofrom.margin_flag);
        setSecu_status(bofrom.secu_status);
        setMemo(bofrom.memo);
        setUpdate_time(bofrom.update_time);
        setUuid(bofrom.uuid);
        setBoid(bofrom.boid);
        setDestination(bofrom.destination);
    }

    public byte[] serial() {
        return builder.build().toByteArray();
    }

    @Override
    public com.google.protobuf.GeneratedMessageV3.Builder getBuilder() { 
        return builder;
    }

    @Override
    public RDSStockBO getBo() { 
        return bo;
    }

    public java.lang.String getProduct_id() {
        return builder.getProductId();
    }

    public void setProduct_id(java.lang.String product_id) {
        bo.product_id = product_id;
        builder.setProductId(product_id);
    }

    public java.lang.String getSecu_name_cn() {
        return builder.getSecuNameCn();
    }

    public void setSecu_name_cn(java.lang.String secu_name_cn) {
        bo.secu_name_cn = secu_name_cn;
        builder.setSecuNameCn(secu_name_cn);
    }

    public java.lang.String getSecu_name_en() {
        return builder.getSecuNameEn();
    }

    public void setSecu_name_en(java.lang.String secu_name_en) {
        bo.secu_name_en = secu_name_en;
        builder.setSecuNameEn(secu_name_en);
    }

    public java.lang.String getExch_type() {
        return builder.getExchType();
    }

    public void setExch_type(java.lang.String exch_type) {
        bo.exch_type = exch_type;
        builder.setExchType(exch_type);
    }

    public java.lang.String getSecu_type() {
        return builder.getSecuType();
    }

    public void setSecu_type(java.lang.String secu_type) {
        bo.secu_type = secu_type;
        builder.setSecuType(secu_type);
    }

    public java.lang.String getSecu_sub_type() {
        return builder.getSecuSubType();
    }

    public void setSecu_sub_type(java.lang.String secu_sub_type) {
        bo.secu_sub_type = secu_sub_type;
        builder.setSecuSubType(secu_sub_type);
    }

    public java.lang.String getCurrency() {
        return builder.getCurrency();
    }

    public void setCurrency(java.lang.String currency) {
        bo.currency = currency;
        builder.setCurrency(currency);
    }

    public java.lang.String getList_date() {
        return builder.getListDate();
    }

    public void setList_date(java.lang.String list_date) {
        bo.list_date = list_date;
        builder.setListDate(list_date);
    }

    public int getBuy_unit() {
        return builder.getBuyUnit();
    }

    public void setBuy_unit(int buy_unit) {
        bo.buy_unit = buy_unit;
        builder.setBuyUnit(buy_unit);
    }

    public int getSell_unit() {
        return builder.getSellUnit();
    }

    public void setSell_unit(int sell_unit) {
        bo.sell_unit = sell_unit;
        builder.setSellUnit(sell_unit);
    }

    public int getTrade_low_limit() {
        return builder.getTradeLowLimit();
    }

    public void setTrade_low_limit(int trade_low_limit) {
        bo.trade_low_limit = trade_low_limit;
        builder.setTradeLowLimit(trade_low_limit);
    }

    public int getTrade_high_limit() {
        return builder.getTradeHighLimit();
    }

    public void setTrade_high_limit(int trade_high_limit) {
        bo.trade_high_limit = trade_high_limit;
        builder.setTradeHighLimit(trade_high_limit);
    }

    public double getPre_close_price() {
        return builder.getPreClosePrice();
    }

    public void setPre_close_price(double pre_close_price) {
        bo.pre_close_price = pre_close_price;
        builder.setPreClosePrice(pre_close_price);
    }

    public double getTick_price() {
        return builder.getTickPrice();
    }

    public void setTick_price(double tick_price) {
        bo.tick_price = tick_price;
        builder.setTickPrice(tick_price);
    }

    public double getPrice_high_limit() {
        return builder.getPriceHighLimit();
    }

    public void setPrice_high_limit(double price_high_limit) {
        bo.price_high_limit = price_high_limit;
        builder.setPriceHighLimit(price_high_limit);
    }

    public double getPrice_low_limit() {
        return builder.getPriceLowLimit();
    }

    public void setPrice_low_limit(double price_low_limit) {
        bo.price_low_limit = price_low_limit;
        builder.setPriceLowLimit(price_low_limit);
    }

    public double getEx_right_ratio() {
        return builder.getExRightRatio();
    }

    public void setEx_right_ratio(double ex_right_ratio) {
        bo.ex_right_ratio = ex_right_ratio;
        builder.setExRightRatio(ex_right_ratio);
    }

    public double getDividend_price() {
        return builder.getDividendPrice();
    }

    public void setDividend_price(double dividend_price) {
        bo.dividend_price = dividend_price;
        builder.setDividendPrice(dividend_price);
    }

    public java.lang.String getFinancing_flag() {
        return builder.getFinancingFlag();
    }

    public void setFinancing_flag(java.lang.String financing_flag) {
        bo.financing_flag = financing_flag;
        builder.setFinancingFlag(financing_flag);
    }

    public java.lang.String getMargin_flag() {
        return builder.getMarginFlag();
    }

    public void setMargin_flag(java.lang.String margin_flag) {
        bo.margin_flag = margin_flag;
        builder.setMarginFlag(margin_flag);
    }

    public java.lang.String getSecu_status() {
        return builder.getSecuStatus();
    }

    public void setSecu_status(java.lang.String secu_status) {
        bo.secu_status = secu_status;
        builder.setSecuStatus(secu_status);
    }

    public java.lang.String getMemo() {
        return builder.getMemo();
    }

    public void setMemo(java.lang.String memo) {
        bo.memo = memo;
        builder.setMemo(memo);
    }

    public java.lang.String getUpdate_time() {
        return builder.getUpdateTime();
    }

    public void setUpdate_time(java.lang.String update_time) {
        bo.update_time = update_time;
        builder.setUpdateTime(update_time);
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public String toString() {
        return "RDSStockBO_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "product_id = " + getProduct_id() +"," +
            "secu_name_cn = " + getSecu_name_cn() +"," +
            "secu_name_en = " + getSecu_name_en() +"," +
            "exch_type = " + getExch_type() +"," +
            "secu_type = " + getSecu_type() +"," +
            "secu_sub_type = " + getSecu_sub_type() +"," +
            "currency = " + getCurrency() +"," +
            "list_date = " + getList_date() +"," +
            "buy_unit = " + getBuy_unit() +"," +
            "sell_unit = " + getSell_unit() +"," +
            "trade_low_limit = " + getTrade_low_limit() +"," +
            "trade_high_limit = " + getTrade_high_limit() +"," +
            "pre_close_price = " + getPre_close_price() +"," +
            "tick_price = " + getTick_price() +"," +
            "price_high_limit = " + getPrice_high_limit() +"," +
            "price_low_limit = " + getPrice_low_limit() +"," +
            "ex_right_ratio = " + getEx_right_ratio() +"," +
            "dividend_price = " + getDividend_price() +"," +
            "financing_flag = " + getFinancing_flag() +"," +
            "margin_flag = " + getMargin_flag() +"," +
            "secu_status = " + getSecu_status() +"," +
            "memo = " + getMemo() +"," +
            "update_time = " + getUpdate_time() +"," +
         "]";
    }
}

File: fstp-core-bo/python/com/purefun/fstp/core/bo/otw/RDSStockBO_OTW.py
from core.common.ICommon_OTW import ICommon_OTW
from com.purefun.fstp.core.bo.pro import RDSStockBO_pb2
from com.purefun.fstp.core.bo.model.RDSStockBO import RDSStockBO

class RDSStockBO_OTW(ICommon_OTW):

    def __init__(self, byteMsg = None):
        self._bo_pro = RDSStockBO_pb2.RDSStockBO()
        self._bo = RDSStockBO()

        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()

    def __setDataFromBO(self):
        self._bo_pro.product_id = self._bo.product_id
        self._bo_pro.secu_name_cn = self._bo.secu_name_cn
        self._bo_pro.secu_name_en = self._bo.secu_name_en
        self._bo_pro.exch_type = self._bo.exch_type
        self._bo_pro.secu_type = self._bo.secu_type
        self._bo_pro.secu_sub_type = self._bo.secu_sub_type
        self._bo_pro.currency = self._bo.currency
        self._bo_pro.list_date = self._bo.list_date
        self._bo_pro.buy_unit = self._bo.buy_unit
        self._bo_pro.sell_unit = self._bo.sell_unit
        self._bo_pro.trade_low_limit = self._bo.trade_low_limit
        self._bo_pro.trade_high_limit = self._bo.trade_high_limit
        self._bo_pro.pre_close_price = self._bo.pre_close_price
        self._bo_pro.tick_price = self._bo.tick_price
        self._bo_pro.price_high_limit = self._bo.price_high_limit
        self._bo_pro.price_low_limit = self._bo.price_low_limit
        self._bo_pro.ex_right_ratio = self._bo.ex_right_ratio
        self._bo_pro.dividend_price = self._bo.dividend_price
        self._bo_pro.financing_flag = self._bo.financing_flag
        self._bo_pro.margin_flag = self._bo.margin_flag
        self._bo_pro.secu_status = self._bo.secu_status
        self._bo_pro.memo = self._bo.memo
        self._bo_pro.update_time = self._bo.update_time
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.destination = self._bo.destination

    def __setDataFromPB(self):
        self._bo.product_id = self._bo_pro.product_id
        self._bo.secu_name_cn = self._bo_pro.secu_name_cn
        self._bo.secu_name_en = self._bo_pro.secu_name_en
        self._bo.exch_type = self._bo_pro.exch_type
        self._bo.secu_type = self._bo_pro.secu_type
        self._bo.secu_sub_type = self._bo_pro.secu_sub_type
        self._bo.currency = self._bo_pro.currency
        self._bo.list_date = self._bo_pro.list_date
        self._bo.buy_unit = self._bo_pro.buy_unit
        self._bo.sell_unit = self._bo_pro.sell_unit
        self._bo.trade_low_limit = self._bo_pro.trade_low_limit
        self._bo.trade_high_limit = self._bo_pro.trade_high_limit
        self._bo.pre_close_price = self._bo_pro.pre_close_price
        self._bo.tick_price = self._bo_pro.tick_price
        self._bo.price_high_limit = self._bo_pro.price_high_limit
        self._bo.price_low_limit = self._bo_pro.price_low_limit
        self._bo.ex_right_ratio = self._bo_pro.ex_right_ratio
        self._bo.dividend_price = self._bo_pro.dividend_price
        self._bo.financing_flag = self._bo_pro.financing_flag
        self._bo.margin_flag = self._bo_pro.margin_flag
        self._bo.secu_status = self._bo_pro.secu_status
        self._bo.memo = self._bo_pro.memo
        self._bo.update_time = self._bo_pro.update_time
        self._bo.uuid = self._bo_pro.uuid
        self._bo.boid = self._bo_pro.boid
        self._bo.destination = self._bo_pro.destination

    def getBO(self):
        return self._bo

    def getProBO(self):
        return self._bo_pro

    def getProduct_id(self):
        return self._bo.product_id

    def setProduct_id(self, product_id):
        self._bo.product_id = product_id
        self._bo_pro.product_id = product_id

    def getSecu_name_cn(self):
        return self._bo.secu_name_cn

    def setSecu_name_cn(self, secu_name_cn):
        self._bo.secu_name_cn = secu_name_cn
        self._bo_pro.secu_name_cn = secu_name_cn

    def getSecu_name_en(self):
        return self._bo.secu_name_en

    def setSecu_name_en(self, secu_name_en):
        self._bo.secu_name_en = secu_name_en
        self._bo_pro.secu_name_en = secu_name_en

    def getExch_type(self):
        return self._bo.exch_type

    def setExch_type(self, exch_type):
        self._bo.exch_type = exch_type
        self._bo_pro.exch_type = exch_type

    def getSecu_type(self):
        return self._bo.secu_type

    def setSecu_type(self, secu_type):
        self._bo.secu_type = secu_type
        self._bo_pro.secu_type = secu_type

    def getSecu_sub_type(self):
        return self._bo.secu_sub_type

    def setSecu_sub_type(self, secu_sub_type):
        self._bo.secu_sub_type = secu_sub_type
        self._bo_pro.secu_sub_type = secu_sub_type

    def getCurrency(self):
        return self._bo.currency

    def setCurrency(self, currency):
        self._bo.currency = currency
        self._bo_pro.currency = currency

    def getList_date(self):
        return self._bo.list_date

    def setList_date(self, list_date):
        self._bo.list_date = list_date
        self._bo_pro.list_date = list_date

    def getBuy_unit(self):
        return self._bo.buy_unit

    def setBuy_unit(self, buy_unit):
        self._bo.buy_unit = buy_unit
        self._bo_pro.buy_unit = buy_unit

    def getSell_unit(self):
        return self._bo.sell_unit

    def setSell_unit(self, sell_unit):
        self._bo.sell_unit = sell_unit
        self._bo_pro.sell_unit = sell_unit

    def getTrade_low_limit(self):
        return self._bo.trade_low_limit

    def setTrade_low_limit(self, trade_low_limit):
        self._bo.trade_low_limit = trade_low_limit
        self._bo_pro.trade_low_limit = trade_low_limit

    def getTrade_high_limit(self):
        return self._bo.trade_high_limit

    def setTrade_high_limit(self, trade_high_limit):
        self._bo.trade_high_limit = trade_high_limit
        self._bo_pro.trade_high_limit = trade_high_limit

    def getPre_close_price(self):
        return self._bo.pre_close_price

    def setPre_close_price(self, pre_close_price):
        self._bo.pre_close_price = pre_close_price
        self._bo_pro.pre_close_price = pre_close_price

    def getTick_price(self):
        return self._bo.tick_price

    def setTick_price(self, tick_price):
        self._bo.tick_price = tick_price
        self._bo_pro.tick_price = tick_price

    def getPrice_high_limit(self):
        return self._bo.price_high_limit

    def setPrice_high_limit(self, price_high_limit):
        self._bo.price_high_limit = price_high_limit
        self._bo_pro.price_high_limit = price_high_limit

    def getPrice_low_limit(self):
        return self._bo.price_low_limit

    def setPrice_low_limit(self, price_low_limit):
        self._bo.price_low_limit = price_low_limit
        self._bo_pro.price_low_limit = price_low_limit

    def getEx_right_ratio(self):
        return self._bo.ex_right_ratio

    def setEx_right_ratio(self, ex_right_ratio):
        self._bo.ex_right_ratio = ex_right_ratio
        self._bo_pro.ex_right_ratio = ex_right_ratio

    def getDividend_price(self):
        return self._bo.dividend_price

    def setDividend_price(self, dividend_price):
        self._bo.dividend_price = dividend_price
        self._bo_pro.dividend_price = dividend_price

    def getFinancing_flag(self):
        return self._bo.financing_flag

    def setFinancing_flag(self, financing_flag):
        self._bo.financing_flag = financing_flag
        self._bo_pro.financing_flag = financing_flag

    def getMargin_flag(self):
        return self._bo.margin_flag

    def setMargin_flag(self, margin_flag):
        self._bo.margin_flag = margin_flag
        self._bo_pro.margin_flag = margin_flag

    def getSecu_status(self):
        return self._bo.secu_status

    def setSecu_status(self, secu_status):
        self._bo.secu_status = secu_status
        self._bo_pro.secu_status = secu_status

    def getMemo(self):
        return self._bo.memo

    def setMemo(self, memo):
        self._bo.memo = memo
        self._bo_pro.memo = memo

    def getUpdate_time(self):
        return self._bo.update_time

    def setUpdate_time(self, update_time):
        self._bo.update_time = update_time
        self._bo_pro.update_time = update_time

    def getUuid(self):
        return self._bo.uuid

    def setUuid(self, uuid):
        self._bo.uuid = uuid
        self._bo_pro.uuid = uuid

    def getBoid(self):
        return self._bo.boid

    def setBoid(self, boid):
        self._bo.boid = boid
        self._bo_pro.boid = boid

    def getDestination(self):
        return self._bo.destination

    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination

    def toString(self):
        return "RDSStockBO_OTW ["+"product_id = " + self.getProduct_id() +"," +"secu_name_cn = " + self.getSecu_name_cn() +"," +"secu_name_en = " + self.getSecu_name_en() +"," +"exch_type = " + self.getExch_type() +"," +"secu_type = " + self.getSecu_type() +"," +"secu_sub_type = " + self.getSecu_sub_type() +"," +"currency = " + self.getCurrency() +"," +"list_date = " + self.getList_date() +"," +"buy_unit = " + str(self.getBuy_unit()) +"," +"sell_unit = " + str(self.getSell_unit()) +"," +"trade_low_limit = " + str(self.getTrade_low_limit()) +"," +"trade_high_limit = " + str(self.getTrade_high_limit()) +"," +"pre_close_price = " + str(self.getPre_close_price()) +"," +"tick_price = " + str(self.getTick_price()) +"," +"price_high_limit = " + str(self.getPrice_high_limit()) +"," +"price_low_limit = " + str(self.getPrice_low_limit()) +"," +"ex_right_ratio = " + str(self.getEx_right_ratio()) +"," +"dividend_price = " + str(self.getDividend_price()) +"," +"financing_flag = " + self.getFinancing_flag() +"," +"margin_flag = " + self.getMargin_flag() +"," +"secu_status = " + self.getSecu_status() +"," +"memo = " + self.getMemo() +"," +"update_time = " + self.getUpdate_time() +"," +"uuid = " + self.getUuid() +"," +"boid = " + str(self.getBoid()) +"," +"destination = " + self.getDestination() +"," +"]"
